openmicroscopy/openmicroscopy
# CLI admin 'session' command

# What this PR does

Adds a ~~'session'~~ 'open' and 'close' command to the CLI ~~admin~~ session plugin, which enables the creation and removal of user sessions.

See https://trello.com/c/I2eVKJGQ/19-rfe-create-session-for-user-cli

Although I'm not sure if that solves the problem of the 'long running' sessions. According to https://downloads.openmicroscopy.org/omero/5.4.0/api/ome/api/ISession.html#createSessionWithTimeout-ome.system.Principal-long- , if one specifies no timeout value (the plugin will then pass on `0`) the maximum timeToLive will be used. I have no idea what this maximum timeToLive actually is.
 
# Testing this PR

- Use the plugin to create a user session. Check with Insight (copy/paste session ID as username and password), that you can connect to the server. 
- Use the `close` command to close the session again. Verify (e.g. with Insight) that the connection is closed.
- Create another session with a short timeout, e. g. 30s. Check that you can connect using the session ID, wait a bit, check again that you cannot connect with the session id any longer.
- Check `test_sessions.py` integration tests
